rust/scx_utils/src/topology.rs

```diff
@@ -307,6 +336,62 @@ impl Topology {
         }
         sibling_cpu
     }
+
+    /// Returns whether AMD preferred core ranking is enabled on this system
+    pub fn has_pref_rank(&self) -> bool {
+        if !Path::new("/sys/devices/system/cpu/amd_pstate/prefcore").exists() {
+            return false;
+        }
+        match std::fs::read_to_string("/sys/devices/system/cpu/amd_pstate/prefcore") {
+            Ok(contents) => contents.trim() == "enabled",
+            Err(_) => false,
+        }
+    }
+
+    /// Returns a sorted list of CPU IDs from highest to lowest rank.
+    /// The list is cached internally and refreshed every 10 seconds.
+    /// If preferred core ranking is not enabled, returns an empty slice.
```

Comment:
rather than running every 10 seconds, how about using that netlink event I suggested?
https://github.com/torvalds/linux/blob/v6.13/drivers/acpi/processor_driver.c#L86